eva-server/src/test/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2IntegrationTest.java

```diff
@@ -136,4 +140,37 @@ public void testGetVariantsByNonExistingRegion() throws URISyntaxException {
     public void testGetVariantsByNonExistingRegions() throws URISyntaxException {
         testGetVariantsByRegionHelper("21:8000-9000,21:8000-9000", 0, HttpStatus.NOT_FOUND);
     }
+
+    @Test
+    public void testPagination() {
+        String url = "/v2/regions/20:60000-62000/variants?species=mmusculus&assembly=grcm38?&page=0&size=10";
+        ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+
+        Configuration configuration = Configuration.defaultConfiguration()
+                .jsonProvider(new JacksonJsonProvider())
+                .mappingProvider(new JacksonMappingProvider(objectMapper))
+                .addOptions(Option.SUPPRESS_EXCEPTIONS);
+        Integer totalNumberOfElements = JsonPath.using(configuration).parse(response.getBody())
+                .read("$['page']['totalElements']", Integer.class);
+        Integer pageNumber = JsonPath.using(configuration).parse(response.getBody())
+                .read("$['page']['number']", Integer.class);
+        Integer size = JsonPath.using(configuration).parse(response.getBody())
+                .read("$['page']['size']", Integer.class);
+        Integer totalPages = JsonPath.using(configuration).parse(response.getBody())
+                .read("$['page']['totalPages']", Integer.class);
+
+        assertEquals(1, totalNumberOfElements.intValue());
+        assertEquals(0, pageNumber.intValue());
+        assertEquals(10, size.intValue());
+        assertEquals(1, totalPages.intValue());
+    }
+
+    @Test
+    public void testInvalidPageRanges() {
+        String url = "/v2/regions/20:60000-62000/variants?species=mmusculus&assembly=grcm38?&page=1000&size=1";
+        ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+        assertEquals(HttpStatus.REQUESTED_RANGE_NOT_SATISFIABLE, response.getStatusCode());
+        assertEquals("The correct page range is from 0 to 0 for the given page size", response.getBody());
```

Comment:
this is perfectly correct, but looks suspicious. can you change it so that it shows like this?
```suggestion
        assertEquals("For the given page size, there are 1 page(s), so the correct page range is from 0 to 0 (both included).", response.getBody());
```